packages/fx-core/src/plugins/resource/bot/v2/bicep.ts

```diff
@@ -0,0 +1,162 @@
+        hostName: functionHostName,
+        functionEndpoint: functionEndpoint,
+        endpointAsParam: endpointAsParam,
+      },
+    };
+    return result;
+  }
+
+  async updateBicep(ctx: Context, configuration: Configurations): Promise<ResourceTemplate> {
+    return {} as ArmTemplateResult;
+  }
+}
+
+export class BotServiceBicepGenerator implements BicepGenerator {
+  async generateBicep(ctx: Context, configuration: Configurations): Promise<ResourceTemplate> {
+    const plugins = getActivatedV2ResourcePlugins(ctx.projectSetting).map(
+      (p) => new NamedArmResourcePluginAdaptor(p)
+    );
+
+    const pluginCtx = { plugins: plugins.map((obj) => obj.name) };
+    const bicepTemplateDir = path.join(
+      getTemplatesFolder(),
+      BicepTemplateRelativeDir,
+      BotServiceBicepFolderName
+    );
+    const bicepFilenames = [Bicep.ProvisionFileName, "botserviceProvision.template.bicep"];
+
+    const modules = await Promise.all(
+      bicepFilenames.map((name) =>
+        generateBicepFromFile(path.join(bicepTemplateDir, name), pluginCtx)
+      )
+    );
+    const result: ArmTemplateResult = {
+      Provision: {
+        Orchestration: modules[0],
+        Modules: { bot: modules[1] },
+      },
+      Parameters: await fs.readJson(path.join(bicepTemplateDir, Bicep.ParameterFileName)),
+    };
+    return result;
+  }
+
+  async updateBicep(ctx: Context, configuration: Configurations): Promise<ResourceTemplate> {
+    return {} as ArmTemplateResult;
+  }
+}
+
+export function getGenerators(ctx: Context, inputs: Inputs): BicepGenerator[] {
+  const generators = [];
+  const rawHostType = ctx.projectSetting?.pluginSettings?.[PluginBot.PLUGIN_NAME]?.[
+    PluginBot.HOST_TYPE
+  ] as string;
+
+  const hostType = utils.convertToConstValues(rawHostType, HostTypes);
+
+  if (hostType === BotHostTypes.AppService) {
+    generators.push(new WebAppBicepGenerator());
+  } else if (hostType === BotHostTypes.AzureFunctions) {
+    generators.push(new FunctionBicepGenerator());
+  }
+
+  generators.push(new BotServiceBicepGenerator());
+  return generators;
+}
+
+export function mergeTemplates(templates: ArmTemplateResult[]): ArmTemplateResult {
+  const result: ArmTemplateResult = {
+    Provision: {
+      Orchestration: templates.map((template) => template.Provision?.Orchestration).join(""),
+      Modules: templates
+        .map((template) => template.Provision?.Modules)
+        .reduce((result, current) => Object.assign(result, current), {}),
+    },
+    Configuration: {
+      Orchestration: templates.map((template) => template.Configuration?.Orchestration).join(""),
+      Modules: templates
+        .map((template) => template.Configuration?.Modules)
+        .reduce((result, current) => Object.assign(result, current), {}),
+    },
+    Parameters: Object.assign({}, ...templates.map((template) => template.Parameters)),
+    Reference: Object.assign({}, ...templates.map((template) => template.Reference)),
```

Comment:
I remember we have some learning on the trap of Object.assign(), just confirm if it works as we expect here?